src/scales/scale.time.js

```diff
@@ -36,10 +40,94 @@ module.exports = function(Chart) {
 			},
 		},
 		ticks: {
-			autoSkip: false
+			autoSkip: false,
+			mode: 'series',   // 'linear|series'
+			source: 'auto'    // 'auto|labels'
 		}
 	};
 
+	function buildLookupTable(ticks, min, max, linear) {
+		var ilen = ticks.length;
+		var table = [];
+		var i, prev, curr, next, decimal;
+
+		if (ilen === 0) {
+			return table;
+		}
+
+		if (ticks[0] !== min) {
+			table.push({time: min, decimal: 0});
+		}
+
+		for (i=0, ilen=ticks.length; i<ilen; ++i) {
+			next = ticks[i+1] || 0;
+			prev = ticks[i-1] || 0;
+			curr = ticks[i];
+
+			// only add points that breaks the scale linearity
+			if (Math.round((next+prev)/2) !== curr) {
+				decimal = linear? (curr - min) / (max - min) : ilen > 1? i/(ilen-1) : 0;
+				table.push({time: curr, decimal: decimal});
+			}
+		}
+
+		if (ticks[ilen-1] !== max) {
+			table.push({time: max, decimal: 1});
+		}
+
+		return table;
+	}
+
+	// @see adapted from http://www.anujgakhar.com/2014/03/01/binary-search-in-javascript/
+	function lookup(table, key, value) {
+		var lo = 0;
+		var hi = table.length -1;
```

Comment:
spacing with operators is inconsistent throughout this function. e.g. there's a space before the `-` but not after